src/shared/elf-util.c

```diff
@@ -624,8 +624,8 @@ static int parse_elf(int fd, const char *executable, char **ret, JsonVariant **r
                 if (r < 0)
                         return log_warning_errno(r, "Failed to parse package metadata of ELF file: %m");
 
-                /* If we found a build-id and nothing else, return at least that. */
-                if (!package_metadata && id_json) {
```

Comment:
I don't think this is right? It's possible nothing is found (it's not an error condition)

Reply:
Here, `package_metadata` is always NULL. I just dropped the redundant condition.

Reply:
It's not always null? Otherwise `parse_package_metadata` would be doing nothing

Reply:
Ah, `package_metadata` is also passed to StackContext... Ouch.